gazelle_cabal/dependency_resolution.go

```diff
@@ -224,42 +223,85 @@ func getPackageLabel(
 	pkgName string,
 	from label.Label,
 ) label.Label {
-	parentPackage, packageName := resolvePackageName(pkgName)
+	var resolvers []func(*resolve.RuleIndex, string, string, label.Label)(label.Label, error)
+	resolvers = append(resolvers,
+		resolveAgainstPrivateLibraries,
+		resolveAgainstPublicLibraries,
+		resolveAgainstRepository,
+	)
+
+	for _, resolver := range resolvers {
+		lbl, err := resolver(ix, packageRepo, pkgName, from)
+		if (err == nil) {
+			return lbl
+		}
+	}
+
+	log.Fatalf("Could not resolve dependency '%s' from '%s'", pkgName, from)
+	return label.Label{}
+}
+
+func resolveAgainstPrivateLibraries(
+	ix *resolve.RuleIndex,
+	packageRepo string,
+	pkgName string,
+	from label.Label,
+) (label.Label, error) {
+	parentPackage, depName := splitDependencyString(pkgName)
+
+	if len(parentPackage) > 0 && (parentPackage != depName) {
```

Comment:
Why should `parentPackage` be `depName`? That appears to be the only way in which the `if` statement can be skipped.

I'd say the library is internal iff `parentPackage` is the same as the name of the rule for which dependencies are being resolved.

Does cabal accept an internal library to be referred as `package-a:lib` from other artifacts of `package-a`?